utest/utils/test_toc.py

```diff
@@ -0,0 +1,103 @@
+import unittest
+
+from robot.utils.table_of_content import add_toc, _get_headings, _create_toc
+
+EXAMPLE_DOC_TOC = """Simple library
+
+%TABLE_OF_CONTENTS%
+
+= 1 Heading =
+
+Some text here for chapter 1.
+
+= 2 Heading =
+
+Second chapter text there.
+
+Second paragraph.
+"""
+
+EXAMPLE_DOC_NO_HEADING = """Simple library
+
+%TABLE_OF_CONTENTS%
+
+Some text here for library.
+
+= This is not heading
+
+No heading here =
+
+This = is not = heading
+
+Other text
+"""
+
+
+class TestTableOfContentLibdoc(unittest.TestCase):
+
+    def test_no_toc(self):
+        test_doc = 'Library doc'
+        doc = add_toc(test_doc, True)
+        self.assertEqual(doc, test_doc, 'Doc should not have been created.')
```

Comment:
We've typically used our own `robot.utils.asserts.assert_equal` to avoid `self.` and using camelCase. Using the standard `self.assertEqual` is fine as well, though.